src/main/java/com/faforever/client/leaderboard/LeaderboardController.java

```diff
@@ -0,0 +1,380 @@
+          newTable.scrollTo(entryToSelect);
+          newTable.getSelectionModel().select(entryToSelect);
+        }));
+  }
+
+  private void plotDivisionDistributions(List<SubdivisionBean> subdivisions, LeagueEntryBean leagueEntry) {
+    JavaFxUtil.runLater(() -> ratingDistributionChart.getData().clear());
+    // We need to set the categories first, to ensure they have the right order.
+    ObservableList<String> categories = FXCollections.observableArrayList();
+    subdivisions.stream()
+        .filter(subdivision -> subdivision.getIndex() == 1)
+        .map(subdivision -> i18n.get(subdivision.getDivisionI18nKey()))
+        .forEach(categories::add);
+    xAxis.setCategories(categories);
+
+    subdivisions.stream().filter(subdivision -> subdivision.getDivision().getIndex() == 1).forEach(firstTierSubDivision -> {
+      XYChart.Series<String, Integer> series = new XYChart.Series<>();
+      series.setName(firstTierSubDivision.getNameKey());
+      subdivisions.stream()
+          .filter(subdivision -> subdivision.getIndex() == firstTierSubDivision.getIndex())
+          .forEach(subdivision -> leaderboardService.getSizeOfDivision(subdivision).thenAccept(size -> {
+            XYChart.Data<String, Integer> data = new XYChart.Data<>(i18n.get(subdivision.getDivisionI18nKey()), size);
+            Text label = new Text();
+            label.setText(subdivision.getNameKey());
+            label.setFill(Color.WHITE);
+            data.nodeProperty().addListener((observable, oldValue, newValue) -> {
+              if (leagueEntry != null && subdivision.equals(leagueEntry.getSubdivision())) {
+                newValue.pseudoClassStateChanged(NOTIFICATION_HIGHLIGHTED_PSEUDO_CLASS, true);
+              }
+              addNodeOnTopOfBar(data, label);
+            });
+            JavaFxUtil.runLater(() -> series.getData().add(data));
+          }));
+      JavaFxUtil.runLater(() -> ratingDistributionChart.getData().add(series));
+    });
+    leaderboardService.getTotalPlayers(season).thenAccept(totalPlayers ->
+        JavaFxUtil.runLater(() -> xAxis.labelProperty().setValue(i18n.get("leaderboard.totalPlayers", totalPlayers))));
+  }
+
+  private void addNodeOnTopOfBar(XYChart.Data<String, Integer> data, Node nodeToAdd) {
+    final Node node = data.getNode();
+    node.parentProperty().addListener((ov, oldParent, parent) -> {
+      if (parent == null) {
+        return;
+      }
+      Group parentGroup = (Group) parent;
+      ObservableList<Node> children = parentGroup.getChildren();
+      if (!children.contains(nodeToAdd)) {
+        children.add(nodeToAdd);
+        nodeToAdd.setViewOrder(-0.5);
+      }
+    });
+
+    JavaFxUtil.addListener(node.boundsInParentProperty(), (ov, oldBounds, bounds) -> {
+      nodeToAdd.setLayoutX(Math.round(bounds.getMinX() + bounds.getWidth() / 2 - nodeToAdd.prefWidth(-1) / 2));
+      nodeToAdd.setLayoutY(Math.round(bounds.getMaxY() - nodeToAdd.prefHeight(-1) * 0.5));
+    });
```

Comment:
I would almost make this its own class but maybe not necessary

Reply:
If we at some point want to have more bar charts like this, then sure. Right now I don't see the use case, so it doesn't seem to be worth the effort

Reply:
That makes sense